dist2src/core.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/python3
+
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+import logging
+import os
+from pathlib import Path
+from typing import Optional, Dict, Any
+
+import git
+import sh
+
+
+logger = logging.getLogger(__name__)
+
+POST_CLONE_HOOK = "post-clone"
+AFTER_PREP_HOOK = "after-prep"
+INCLUDE_SOURCES = "include-files"
+
+# would be better to have them externally (in a file at least)
+# but since this is only for kernel, it should be good enough
+KERNEL_DEBRAND_PATCH_MESSAGE = """\
+Debranding CPU patch
+
+present_in_specfile: true
+location_in_specfile: 1000
+patch_name: debrand-single-cpu.patch
+"""
+HOOKS: Dict[str, Dict[str, Any]] = {
+    "kernel": {
+        # %setup -c creates another directory level but patches don't expect it
+        AFTER_PREP_HOOK: (
+            "set -e; "
+            "shopt -s dotglob nullglob && "  # so that * would match dotfiles as well
+            "cd BUILD/kernel-4.18.0-*.el8/ && "
+            "mv ./linux-4.18.0-*.el8.x86_64/* . && "
+            "rmdir ./linux-4.18.0-*.el8.x86_64 && "
+            "git add . && "
+            "git commit --amend --no-edit"
+            # the patch is already applied in %prep
+            # "git apply ../../SOURCES/debrand-single-cpu.patch &&"
+            # f"git commit -a -m '{KERNEL_DEBRAND_PATCH_MESSAGE}'"
+        )
+    },
+    "pacemaker": {INCLUDE_SOURCES: ["nagios-agents-metadata-*.tar.gz"]},
+}
+
+
+def get_hook(package_name: str, hook_name: str) -> Optional[str]:
+    """ get a hook's command for particular source-git repo """
+    return HOOKS.get(package_name, {}).get(hook_name, None)
+
+
+class GitRepo:
+    """
+    a wrapper on top of git.Repo for our convenience
+    """
```

Comment:
We should merge this one with `LocalProject`. Or move the relevant part from LocalProject to a common class.

Nothing that needs to be (re)solved here. Just an idea to finally improve `LocalProject`..;)

Reply:
+1

I'd love to have LocalProject either as a dedicated project or included in ogr :)

Reply:
WE SHOULD PRINT T-SHIRTS WITH LocalProject

Reply:
![i-contributed-to-localproject](https://user-images.githubusercontent.com/1662493/91310061-afa0cc80-e7b1-11ea-9640-fb2075155534.png)